### PR TITLE
Apply --set flags using tpath instead of building tree

### PR DESCRIPTION
**Pull request by ostromart**
This changes the approach for applying set flags from constructing a YAML tree from the flags and overlaying it to using tpath traversal instead. The advantage is that merges involving lists are done correctly which allows more robust handling of paths with list selection.

### CODE REVIEW

File: operator/cmd/mesh/profile-common.go [hunk 1]
```diff
@@ -119,32 +121,20 @@ func parseYAMLFiles(inFilenames []string, force bool, l clog.Logger) (overlayYAM
...
 // files and the --set flag. If successful, it returns an IstioOperatorSpec string and struct.
-func genIOPSFromProfile(profileOrPath, fileOverlayYAML, setOverlayYAML string, skipValidation bool,
+func genIOPSFromProfile(profileOrPath, fileOverlayYAML string, setFlags []string, skipValidation bool,
 	kubeConfig *rest.Config, l clog.Logger) (string, *v1alpha1.IstioOperatorSpec, error) {
-	userOverlayYAML, err := util.OverlayYAML(fileOverlayYAML, setOverlayYAML)
-	if err != nil {
-		return "", nil, fmt.Errorf("could not merge file and --set YAMLs: %s", err)
-	}
-	installPackagePath, err := getInstallPackagePath(userOverlayYAML)
+	//userOverlayYAML, err := util.OverlayYAML(fileOverlayYAML, setOverlayYAML)
```
Comment (howardjohn) [inline operator/cmd/mesh/profile-common.go:128]: do we still need this?

File: operator/cmd/mesh/testdata/manifest-generate/output/gateways.golden.yaml [hunk 2]
```diff
@@ -203,7 +203,7 @@ spec:
             cpu: 2000m
             memory: 1024Mi
           requests:
-            cpu: 111m
+            cpu: 999m
```
Comment (howardjohn) [inline operator/cmd/mesh/testdata/manifest-generate/output/gateways.golden.yaml:206]: This is suspicious. Is this a bug fix or a bug?

Comment (ostromart) [reply operator/cmd/mesh/testdata/manifest-generate/output/gateways.golden.yaml:206]: it's a new test case. the setting would be 111m using the overlay but the --set flag overrides it to 999m. it simultaneously tests the set flag and overlay priority.

File: operator/cmd/mesh/profile-common.go [hunk 3]
```diff
@@ -20,11 +20,10 @@ import (
 	"path/filepath"
 	"strings"
 
+	yaml2 "github.com/ghodss/yaml"
```
Comment (richardwxn) [inline operator/cmd/mesh/profile-common.go:23]: do we use other yaml package here?

File: operator/cmd/mesh/profile-common.go (hunk 1 above)
Comment (ostromart) [reply operator/cmd/mesh/profile-common.go]: removed.

File: operator/cmd/mesh/profile-common.go (hunk 3 above)
Comment (ostromart) [reply operator/cmd/mesh/profile-common.go]: no, it doesn't matter, restored the old one.

**Review by richardwxn — APPROVED** [review]

**Review by esnible — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**